src/hb-ot-name-table.hh

```diff
@@ -169,16 +169,31 @@ _hb_ot_name_entry_cmp_key (const void *pa, const void *pb)
   if (a->language == b->language) return 0;
   if (!a->language) return -1;
   if (!b->language) return +1;
-  return strcmp (hb_language_to_string (a->language),
-		 hb_language_to_string (b->language));
+
+  const char *astr = hb_language_to_string (a->language);
+  const char *bstr = hb_language_to_string (b->language);
+
+  signed c = strcmp (astr, bstr);
+
+  if (!exact && c)
+  {
+    unsigned la = strlen (astr);
+    unsigned lb = strlen (bstr);
+    // 'a' is the user request, and 'b' is string in the font.
+    // If eg. user asks for "en-us" and font has "en", approve.
+    if (la > lb && astr[lb] == '-')
```

Comment:
You don't actually check that the prefixes match here ? Needs a strncmp (astr, bstr, lb) == 0

It doesn't matter for these short strings, but you could write that check without entirely iterating both strings twice,

Reply:
Oops right. Fixed using the slow strncmp version.